### PR TITLE
CWE-79 small tightening up

### PR DESCRIPTION
**Pull request by akalenyu**
**What this PR does / why we need it**:
The uploadReady func **could** be returning an error that's somehow dependent on user input.
It's somehow unlikely and an edge case but better safe than sorry

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by awels** [comment]
looks like this change is affecting a particular XSS test?

File: pkg/uploadproxy/uploadproxy.go [hunk 1]
```diff
@@ -252,8 +252,8 @@ func (app *uploadProxyApp) resolveUploadPath(pvc *v1.PersistentVolumeClaim, pvcN
 			path = common.UploadArchivePath
 		}
 	default:
-		// Escaping user-controlled strings to avoid cross-site scripting (XSS) attacks
-		return "", fmt.Errorf("rejecting upload request for PVC %s - upload content-type %s is invalid", html.EscapeString(pvcName), html.EscapeString(contentType))
+		// Caller is escaping user-controlled strings to avoid cross-site scripting (XSS) attacks
```
Comment (alromeros) [inline pkg/uploadproxy/uploadproxy.go:255]: Nit: Since we are keeping this comment it may make more sense to move it where we are escaping the strings.

Comment (akalenyu) [reply pkg/uploadproxy/uploadproxy.go:255]: I changed it a bit with the intention to keep it here, but I don't mind. Did you catch the wording change?

Comment (alromeros) [reply pkg/uploadproxy/uploadproxy.go:255]: Waiting for the tests, I'll lgtm once they pass

Comment (akalenyu) [reply pkg/uploadproxy/uploadproxy.go:255]: Yeah this time I ran them locally :stuck_out_tongue:

**Pull request merged**